### DISCUSSION TITLE
Chapter 5, instructions broken?

### DISCUSSION THREAD

**Discussion by valentinogagliardi** [browserengineering/book#1347] [Q&A]
Thanks for this wonderful book! 

In chapter 5, section 1, after following all the steps, you're left with a broken implementation, even though it says "test your browser and make sure that after all of these refactors, everything still works."

After applying the changes, the user is left with:

```python
AttributeError: 'BlockLayout' object has no attribute 'layout'
```

`BlockLayout.layout` gets not implemented until [section 2](https://browser.engineering/layout.html#block-layout).

**Comment by pavpanchekha** [comment]
You're intended to rename the constructor of the old `Layout` class to a new `layout` method. There's not a code block for it but it's mentioned in the text.
